### ISSUE TITLE
Fine tuning/transfer learning in YOLOV4

### ISSUE THREAD

**Issue by DrMRizk** [AlexeyAB/darknet#7870]
### Discussed in https://github.com/AlexeyAB/darknet/discussions/7869

<div type='discussions-op-text'>

<sup>Originally posted by **DrMRizk** July  6, 2021</sup>
I trained a network targeting a custom dataset including C classes. I need to train the network with additional images in order to  increase the accuracy. I do not need to increase the number of classes. 
I aim to re-train the network on edge devices taking into consideration low computational resources and power consumption.   

I prepared the dataset including several images of all classes.
Then, I train the network using the custom training command:
./darknet detector train data_file cfg_file yolo4.conv.137 -dont_show -map
for sure the data_file includes the paths to the images/lables for training datset and validation dataset in addition to the path to save the new weights file and the classes names
also, the cfg_file is edited according to the number of classes by changing the max_batches and  filters for layers the comes before the yolo layers

After finishing the training on custom dataset, additional images are prepared.

I need to launch the retraining process to enhance the accuracy; 
using the following command does nothing:
./darknet detector train data_file cfg_file custom_weights_file -dont_show -map 
even when adding stopbackward=1
where custom_weights_file is the output of my initial custom training

I checked the discussions on this  Repository:

I guess I am dealing with issue of  fine tuning as I am using pre-trained weights file from same task. The following is suggested: 

1- use partial to copy the weights of the selected layers from previous obtained weights file.
2- train starting from a pre-trained  weight

in my case, I have to use the following command:
./darknet partial cfg_file selected_weights_file weights_file 137
to select the copy the weights of the first 137 layers from the obtained custom_weights_file from my initial custom training.

then I have to use the  weight file including the copied weights (selected_weights_file) to launch the retraining process. I shall use this command:

./darknet detector train data_file cfg_file selected_weights_file

a- is up listed procedure correct?
b- shall I use only the new images (not used before in training)  or I have to use all images (add the new images to the old image set) in the re-traing process
c- if the new images belong to the same class. would I have degradation because of over-fitting? 
d- what is the minimal number of new images should be added?
e- shall I add stopbackward=1 to the cfg_file and what is its affect in my case?

@AlexeyAB 
@WongKinYiu 

additional image</div>

**Comment by akashAD98** [comment]
@DrMRizk what's your approach?? I'm also dealing with the same issue

**Comment by erminkev1** [comment]
I have similar questions really, thank you for phrasing them so nicely.

I think that I could contribute slightly to the understanding of transfer learning / fine tuning here. If the network is fine-tuned, then the idea is to let all the weights update, without frozen layers, however with a lower learning rate. 

That being said, adding a stopbackward is not necessary to your cfg file.
